### PR TITLE
Add support for processing Unified TLS frames

### LINKED ISSUE #68310: CertificateSelection callback is not called when client use Unified framing
Some older TLS implementations can use Unified framing e.g. legacy SSL2 format with TLS 1.x body. 
With 6.0, we fail to detect the client hello and therefore we fail to call the callback.
This is going to fail completely with 7.0 code base.

### PR DESCRIPTION
**Pull request by wfurt**
This is mix of https://github.com/dotnet/runtime/pull/68425, https://github.com/microsoft/reverse-proxy/pull/1656 and reaction to https://github.com/dotnet/runtime/pull/64322.

Unified parsing works only on Windows as legacy and allows (old) clients to send TLS frame encoded in Ssl2 format.
All we really need is to process first frame. Server should respond back with "normal" TLS e.g. all the framing and processing removed in #64322 is really not needed. 

Since the header size is now variable, I changed `TlsFrameHelper` to return length of the TLS frame instead of just length of the inner payload.

Since we don't have good way how to generate real traffic, I added test that injects legacy ClientHello and verifies that we get back ServerHello instead of failure.  

fixes #68310

### CODE REVIEW

File: src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs [hunk 1]
```diff
@@ -1017,6 +1017,71 @@ public async Task SslStream_RandomSizeWrites_OK(int bufferSize, int readBufferSi
...
+            else
+            {
+                options.ServerCertificateSelectionCallback = (o, name) =>
+                {
+                    callbackCalled = true;
+                    return Configuration.Certificates.GetServerCertificate();
+                };
+
+                serverTask = ssl.AuthenticateAsServerAsync(options, cts.Token);
+            }
```
Comment (rzikm) [inline src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs:1054]: Shouldn't the two branches be swapped?

**Review by rzikm — APPROVED** [review]
LGTM

File: src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs (hunk 1 above)
Comment (wfurt) [reply src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs:1054]: The naming may not be clear but if `useOptionCallback` is `true` we use `AuthenticateAsServerAsync(ServerOptionsSelectionCallback, Object, CancellationToken)` and `AuthenticateAsServerAsync(SslServerAuthenticationOptions, CancellationToken)` in opposite case. 

So there is callback in either branch but we either return whole options or just the certificate. 
Former depends on detecting ClinetHello so it is different code path.

**Comment by wfurt** [comment]
test failures are unrelated to SslStream.

**Pull request merged**